engine/src/main/java/org/hibernate/search/bridge/builtin/time/impl/LocalDateNumericBridge.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.bridge.builtin.time.impl;
+
+import java.time.LocalDate;
+
+/**
+ * Converts a {@link LocalDate} to a {@link Long}.
+ * <p>
+ * The digits of the resulting number will have the following pattern: yyyyMMdd
+ * <p>
+ * For example, the date 2001-3-25 will become the number 20010325.
```

Comment:
> I feel it's very confusing to store this as a numeric value that way,

Why do you think it is confusing?
This solution has the readability of the string approach plus the performance of the numeric approach. Seems the best of the two worlds to me.